integrationtests/tests/fixtures/mod.rs

```diff
@@ -68,124 +69,416 @@ mod real;
+        }
+    }
+    pub fn with_bitcoin(self) -> Self {
+        let bitcoin: Box<dyn BitcoinTest>;
+        let bitcoin_rpc: Box<dyn Fn() -> Box<dyn BitcoindRpc>>;
+
+        if self.disable_mocks {
+            let wallet_config = self.server_config.iter().last().unwrap().1.wallet.clone();
+            bitcoin_rpc = Box::new(bitcoincore_rpc::bitcoind_gen(wallet_config.clone()));
+            bitcoin = Box::new(RealBitcoinTest::new(wallet_config));
+        } else {
+            let fake_bitcoin_test = FakeBitcoinTest::new();
+            bitcoin = Box::new(fake_bitcoin_test.clone());
+            bitcoin_rpc =
+                Box::new(move || Box::new(fake_bitcoin_test.clone()) as Box<dyn BitcoindRpc>);
+        }
+
+        Self {
+            bitcoin: Some(bitcoin),
+            bitcoin_rpc: Some(Box::new(bitcoin_rpc)),
+            ..self
+        }
+    }
+    //TODO: add mock options (e. g fail X times then succeed)
+    pub async fn with_lightning(self) -> Self {
+        let lightning: Box<dyn LightningTest>;
+        let lightning_rpc: Box<dyn LnRpc>;
+
+        if self.disable_mocks {
+            let dir = self
+                .real_dir
+                .as_ref()
+                .expect("disable_mocks is true so the dir got initialized in the constructor");
+
+            let mut socket_gateway = dir.clone();
+            socket_gateway.push("ln1/regtest/lightning-rpc");
+
+            let mut socket_other = dir.clone();
+            socket_other.push("ln2/regtest/lightning-rpc");
+
+            let lightning_prep = RealLightningTest::new(socket_gateway.clone(), socket_other).await;
+            let lightning_rpc_prep = Mutex::new(
+                ClnRpc::new(socket_gateway)
                     .await
                     .expect("connect to ln_socket"),
             );
+            lightning = Box::new(lightning_prep);
+            lightning_rpc = Box::new(lightning_rpc_prep);
+        } else {
+            let lightning_prep = FakeLightningTest::new();
+            lightning_rpc = Box::new(lightning_prep.clone());
+            lightning = Box::new(lightning_prep);
+        }
+
+        Self {
+            lightning: Some(lightning),
+            lightning_rpc: Some(lightning_rpc),
+            ..self
+        }
+    }
+    pub async fn with_gateway(mut self) -> Self {
+        let gateway = GatewayTest::new(
+            self.lightning_rpc
+                .take() // this is a bit weird
+                .expect("self.with_lightning() has been executed and set lightning_rpc"),
+            self.client_config.clone(),
+            self.lightning
+                .as_ref()
+                .expect("self.with_lightning() has been executed and set lightning")
+                .pub_key(),
+            self.base_port + self.num_peers + 1,
+        )
+        .await;
+        Self {
+            gateway: Some(gateway),
+            ..self
+        }
+    }
+    pub async fn with_lightning_and_gateway(mut self) -> Self {
```

Comment:
I will make all `with_xxx` private..the `build` functions will call them anyway (with them beeing public we have the option to say `new().with_xx().with_yy().build()` but becasue we want build to safly unrap everything there is no point in not just doing `new().some_pub_options().build()`